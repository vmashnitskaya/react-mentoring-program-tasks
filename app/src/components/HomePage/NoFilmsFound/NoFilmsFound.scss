@import "../../../constants";

.no-films {
  display: flex;
  flex-direction: column;
  flex: auto;
  justify-content: center;
  align-items: center;
  background-color: $main-dark;
  height: 400px;

  h2 {
    color: $main-white
  }

  button {
    border: 1px solid $main-pink;
    background-color: transparent;
    color: $main-pink;
    padding: 10px 20px;
    margin-top: 30px;
  }
}
