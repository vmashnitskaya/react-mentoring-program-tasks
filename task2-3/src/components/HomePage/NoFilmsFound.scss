@import "../../constants";

.no-films {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: $main-dark;
  height: 300px;

  h2 {
    color: $main-white
  }
}
