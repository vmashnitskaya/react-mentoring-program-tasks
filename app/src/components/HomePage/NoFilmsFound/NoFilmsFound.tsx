import React, { FunctionComponent } from 'react';
import {useHistory} from "react-router-dom";
import './NoFilmsFound.scss';
import Button from "../../common/Button/Button";

const NoFilmsFound: FunctionComponent = () => {
    const history = useHistory();

    const handleBackHome = () => {
        history.push("/home");
    }

    return (
        <div className="no-films">
            <h2>No films found</h2>
            <Button text="Go back home" onClick={handleBackHome}/>
        </div>
    );
};

export default NoFilmsFound;
