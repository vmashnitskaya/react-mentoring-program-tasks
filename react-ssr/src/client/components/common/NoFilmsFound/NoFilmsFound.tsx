import React, { FunctionComponent } from 'react';
import loadable from '@loadable/component';
import {useHistory} from "react-router-dom";
import './NoFilmsFound.scss';

const Button = loadable(() => import('../Button/Button'));

const NoFilmsFound: FunctionComponent = () => {
    const history = useHistory();

    const handleBackHome = () => {
        history.push("/");
    }

    return (
        <div className="no-films">
            <h2>No films found</h2>
            <Button text="Go back home" onClick={handleBackHome}/>
        </div>
    );
};

export default NoFilmsFound;
