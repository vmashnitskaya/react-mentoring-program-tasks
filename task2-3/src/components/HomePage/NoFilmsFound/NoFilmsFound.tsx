import React, { FunctionComponent } from 'react';
import './NoFilmsFound.scss';

const NoFilmsFound: FunctionComponent = () => {
    return (
        <div className="no-films">
            <h2>No films found</h2>
        </div>
    );
};

export default NoFilmsFound;
