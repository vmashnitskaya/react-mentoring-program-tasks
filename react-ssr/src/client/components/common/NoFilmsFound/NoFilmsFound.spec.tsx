import React from 'react';
import { render } from '@testing-library/react';
import NoFilmsFound from "./NoFilmsFound";

describe('No films found', () => {
    test('renders no films found snapshot', () => {
        const {asFragment} = render(<NoFilmsFound />);

        expect(asFragment()).toMatchSnapshot();
    });
});
