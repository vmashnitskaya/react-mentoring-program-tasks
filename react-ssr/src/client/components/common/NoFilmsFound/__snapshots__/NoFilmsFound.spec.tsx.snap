// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No films found renders no films found snapshot 1`] = `
<DocumentFragment>
  <div
    class="no-films"
  >
    <h2>
      No films found
    </h2>
    <button
      class="uppercase"
      type="button"
    >
      Go back home
    </button>
  </div>
</DocumentFragment>
`;
